src/common/templates/_databaseconnections.tpl

```diff
@@ -120,7 +120,20 @@
 {{- $userVariableName := default (printf "%s_USER" $dbType) .userVariableName -}}
 {{- $passwordVariableName := default (printf "%s_PASSWORD" $dbType) .passwordVariableName -}}
 {{- $protocol := (pluck $type .context.Values.global.database | first ).protocol }}
-{{- include "harnesscommon.dbconnection.connection" (dict "type" $type "hosts" $hosts "protocol" $protocol "extraArgs" "/harness" "userVariableName" $userVariableName "passwordVariableName" $passwordVariableName) }}
+{{- $extraArgs:= (pluck $type .context.Values.global.database | first ).extraArgs }}
+{{- $appendedArgs := (printf "/%s" .database )}}
+{{- if or .args $extraArgs }}
+{{- $appendedArgs = (printf "%s?" $appendedArgs)}}
+{{- end }}
+{{- if .args }}
+{{- $appendedArgs = (printf "%s%s" $appendedArgs .args )}}
+{{- end }}
+{{- if and .args $extraArgs }}
+{{- $appendedArgs = (printf "%s&" $appendedArgs)}}
+{{- end }}
+{{- $appendedArgs = (printf "%s%s" $appendedArgs $extraArgs )}}
```

Comment:
```suggestion
{{- $finalArgs := (printf "/%s" .database) }}
{{- if and .args $extraArgs }}
{{- $finalArgs = (printf "%s?%s&%s" $finalArgs .args $extraArgs) }}
{{- else }}
{{- $finalArgs = (printf "%s?%s" $finalArgs (default .args $extraArgs)) }}
{{- end }}
```

Reply:
will not work if both the args are missing

Reply:
just tweak it a bit, the goal is to simplify the logic

Reply:
I tweaked it

Reply:
Done